File: lab1/SimUDuck/ModelDuck.cpp
#include "stdafx.h"
#include "ModelDuck.h"

ModelDuck::ModelDuck()
	: Duck(std::make_unique<FlyNoWay>(), std::make_unique<QuackBehavior>())
{
}

void ModelDuck::Display() const
{
	std::cout << "I'm model duck" << std::endl;
}

File: lw1/src/duck/ModelDuck.php
<?php

class ModelDuck extends Duck
{
    public function __construct()
    {
        parent::__construct(
            new FlyNoWay(),
            new QuackBehavior(),
            new NoDanceBehavior());
    }

    public function display(): void
    {
        echo 'I\'m model duck!' . PHP_EOL;
    }
}